jupyterlite_sphinx/jupyterlite_sphinx.py

```diff
@@ -410,43 +484,58 @@ def run(self):
         )
 
         if self.arguments:
+            # Keep track of the notebooks we are going through, so that we don't
+            # operate on notebooks that are not meant to be included in the built
+            # docs, i.e., those that have not been referenced in the docs via our
+            # directives anywhere.
+            if not hasattr(self.env, "jupyterlite_notebooks"):
+                self.env.jupyterlite_notebooks = set()
+
             # As with other directives like literalinclude, an absolute path is
             # assumed to be relative to the document root, and a relative path
             # is assumed to be relative to the source file
             rel_filename, notebook = self.env.relfn2path(self.arguments[0])
             self.env.note_dependency(rel_filename)
 
-            notebook_name = os.path.basename(notebook)
+            notebook_path = Path(notebook)
 
-            notebooks_dir = Path(self.env.app.srcdir) / CONTENT_DIR / notebook_name
+            self.env.jupyterlite_notebooks.add(str(notebook_path))
+
+            notebooks_dir = Path(self.env.app.srcdir) / CONTENT_DIR
+            os.makedirs(notebooks_dir, exist_ok=True)
+
+            target_name = self._get_target_name(notebook_path, notebooks_dir)
+            target_path = notebooks_dir / target_name
 
             notebook_is_stripped: bool = self.env.config.strip_tagged_cells
 
-            # Create a folder to copy the notebooks to and for NotebookLite to find
-            os.makedirs(os.path.dirname(notebooks_dir), exist_ok=True)
-
-            if notebook_is_stripped:
-                # Note: the directives meant to be stripped must be inside their own
-                # cell so that the cell itself gets removed from the notebook. This
-                # is so that we don't end up removing useful data or directives that
-                # are not meant to be removed.
-
-                nb = nbformat.read(notebook, as_version=4)
-                nb.cells = [
-                    cell
-                    for cell in nb.cells
-                    if "jupyterlite_sphinx_strip" not in cell.metadata.get("tags", [])
-                ]
-                nbformat.write(nb, notebooks_dir, version=4)
-
-            # If notebook_is_stripped is False, then copy the notebook(s) to notebooks_dir.
-            # If it is True, then they have already been copied to notebooks_dir by the
-            # nbformat.write() function above.
+            if notebook_path.suffix.lower() == ".md":
+                if self._should_convert_notebook(notebook_path, target_path):
```

Comment:
```suggestion
                if self._target_is_stale(notebook_path, target_path):
```

corresponds with an earlier suggestion. makes it more clear what's going on in my opinion